packages/suite/src/reducers/wallet/coinjoinReducer.ts

```diff
@@ -241,6 +245,71 @@ const updateClientStatus = (
     };
 };
 
+const appendSessionPhase = (
+    draft: CoinjoinState,
+    payload: ExtractActionPayload<typeof COINJOIN.CLIENT_SESSION_PHASE>,
+) => {
+    const accounts = payload.accountKeys.flatMap(
+        accountKey => draft.accounts.find(({ key }) => key === accountKey) || [],
+    );
+
+    if (!accounts || !accounts.length) {
+        return;
+    }
+
+    accounts.forEach(({ session }) => {
+        if (!session) {
+            return;
+        }
+
+        const isPreviousRoundPhase =
+            getRoundPhaseFromSessionPhase(payload.phase) <
+            getRoundPhaseFromSessionPhase(session?.sessionPhaseQueue.at(-1) || 0);
+        const isSameSessionPhase = payload.phase === session?.sessionPhaseQueue.at(-1);
+
+        const isFirstPhase =
+            getRoundPhaseFromSessionPhase(payload.phase) === RoundPhase.InputRegistration;
+
+        if (isSameSessionPhase || (isPreviousRoundPhase && !isFirstPhase)) {
+            return;
+        }
+
+        const timeSinceLastChange =
+            Date.now() - (session.lastSessionPhaseChangeTimestamp || Date.now());
```

Comment:
doesn't this make the reducer impure? the time may come in the payload from action